packages/clay-data-provider/src/LRUCache.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const DEFAULT_MAX_SIZE = 20;
+
+export class LRUCache<K, V> {
+	maxSize: number;
+	storage: Map<K, V>;
+
+	constructor(maxSize: number = DEFAULT_MAX_SIZE) {
+		if (maxSize <= 0) {
+			throw new TypeError('LRUCache: max must be a non-negative number');
+		}
+
+		this.maxSize = maxSize;
+		this.storage = new Map();
+	}
+
+	reset() {
+		this.storage = new Map();
+		return this.storage;
+	}
+
+	has(key: K) {
+		return this.storage.has(key);
+	}
+
+	get(key: K) {
+		const value = this.storage.get(key);
+
+		if (value) {
```

Comment:
This check is probably not correct, because I could create a cache that stores booleans or other `false`-y values, and those wouldn't get bumped in the cache. That's why in [my implementation](https://github.com/wincent/masochist/blob/8ee6fc3704b204adae6dc3ad9baca5108a741637/src/server/LRUCache.js) that I showed you it does a `storage.has()` check instead.

Reply:
Hmm, I had forgotten about these values ​​that may be here, thanks for this @wincent.